dali/core/alloc_test.cc

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "dali/core/mm/memory.h"
+#include "dali/core/mm/malloc_resource.h"
+#include "dali/core/mm/mm_test_utils.h"
+#include "dali/core/cuda_stream.h"
+
+namespace dali {
+
+TEST(Alloc, Host) {
+  mm::test::test_host_resource mr;
```

Comment:
Does it make sense to add HostDefault test as well?

Reply:
I'll use the default resource in HostShared test.